python/cuml/fil/fil.pyx

```diff
@@ -0,0 +1,341 @@
+        # synchronous w/o a stream
+        return preds
+
+    def load_from_treelite_model(self,
+                                 TreeliteModel model,
+                                 bool output_class,
+                                 algo_t algo,
+                                 float threshold):
+        cdef treelite_params_t treelite_params
+        treelite_params.output_class = output_class
+        treelite_params.threshold = threshold
+        treelite_params.algo = algo
+        self.forest_data = NULL
+        cdef cumlHandle* handle_ =\
+            <cumlHandle*><size_t>self.handle.getHandle()
+        cdef uintptr_t model_ptr = <uintptr_t>model.handle
+        from_treelite(handle_[0],
+                      &self.forest_data,
+                      <ModelHandle> model_ptr,
+                      &treelite_params)
+        return self
+
+    def __cdel__(self):
+        cdef cumlHandle* handle_ =\
+            <cumlHandle*><size_t>self.handle.getHandle()
+        free(handle_[0],
+             self.forest_data)
+        return self
+
+
+class FIL(Base):
+    """
+    Parameters
+    ----------
+    handle : cuml.Handle
+       If it is None, a new one is created just for this class.
+    """
+    def __init__(self,
+                 handle=None):
+        super(FIL, self).__init__(handle)
+        self._impl = FIL_impl(self.handle)
+
+    def predict(self, X, preds=None):
+        """
+        Predicts the labels for X with the loaded forest model
+
+        Parameters
+        ----------
+        X : array-like (device or host) shape = (n_samples, n_features)
+           Dense matrix (floats) of shape (n_samples, n_features).
+           Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
+           ndarray, cuda array interface compliant array like CuPy
+           For optimal performance, pass a device array with C-style layout
+        preds: gpuarray or cudf.Series, shape = (n_samples,)
+           Optional 'out' location to store inference results
+
+        Returns
+        ----------
+        GPU array of length n_samples with inference results
+        (or 'preds' filled with inference results if preds was specified)
+        """
+        return self._impl.predict(X, preds)
+
+    def load_from_treelite_model(self, model, output_class,
+                                 algo=algo_t.TREE_REORG,
+                                 threshold=0.5):
+        """
+        Creates a FIL model using the treelite model
+        passed to the function.
+
+        Parameters
+        ----------
+        model : the model information in the treelite format
+           loaded from a saved model using the treelite API
+           https://treelite.readthedocs.io/en/latest/treelite-api.html
+        output_class: boolean
+           If true, return a 1 or 0 depending on whether the raw prediction
+           exceeds the threshold. If False, just return the raw prediction.
+        algo : int (from algo_t enum)
```

Comment:
Is it possible to only mention the named constants for `algo`, and not their values?
Also, feel free to make it a string parameter.